src/EnergyPlus/ZoneContaminantPredictorCorrector.cc

```diff
@@ -1755,23 +1755,37 @@ namespace ZoneContaminantPredictorCorrector {
                 ControlledGCZoneFlag = false;
                 // Check all the controlled zones to see if it matches the zone simulated
                 for (ContControlledZoneNum = 1; ContControlledZoneNum <= NumContControlledZones; ++ContControlledZoneNum) {
-                    if (GetCurrentScheduleValue(ContaminantControlledZone(ContControlledZoneNum).AvaiSchedPtr) > 0.0) {
-                        ZoneAirGCSetPoint = ZoneGCSetPoint(ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum);
-                        if (ContaminantControlledZone(ContControlledZoneNum).EMSOverrideGCSetPointOn) {
-                            ZoneAirGCSetPoint = ContaminantControlledZone(ContControlledZoneNum).EMSOverrideGCSetPointValue;
+                    if (ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum == ZoneNum) {
+                        if (GetCurrentScheduleValue(ContaminantControlledZone(ContControlledZoneNum).AvaiSchedPtr) > 0.0) {
+                            ZoneAirGCSetPoint = ZoneGCSetPoint(ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum);
+                            if (ContaminantControlledZone(ContControlledZoneNum).EMSOverrideCO2SetPointOn) {
+                                ZoneAirGCSetPoint = ContaminantControlledZone(ContControlledZoneNum).EMSOverrideGCSetPointValue;
+                            }
+                            ControlledGCZoneFlag = true;
+                            break;
```

Comment:
Same comment here about the break statement.  You do, definitely, want to break here, and not continue looping, right?

Reply:
@Myoldmopar Yes.